core/src/main/java/org/bouncycastle/crypto/engines/RFC5649WrapEngine.java

```diff
@@ -154,7 +154,7 @@ public String getAlgorithmName()
             throw new InvalidCipherTextException("unwrap data must be a multiple of 8 bytes");
         }
 
-        if (n == 1)
+        if (n <= 1)
```

Comment:
If the length was zero, then the it was processed as if it was 16 bytes, which caused a runtime exception.